statbot/sql.py

```diff
@@ -955,3 +965,46 @@ def upsert_emoji(self, trans, emoji):
                     )
         trans.execute(ups)
         self.emoji_cache[id] = values
+
+    # Message History
+    def lookup_message_hist(self, trans, channel):
+        self.logger.info(f"Looking up message history for #{channel.name}")
+        sel = select([self.tb_crawl_ranges]) \
+                .where(self.tb_crawl_ranges.c.channel_id == channel.id)
+        result = trans.execute(sel)
+
+        if result.rowcount:
+            _, first, starts, ends = result.fetchone()
+            assert len(starts) == len(ends)
+            ranges = (Range(*r) for r in zip(starts, ends))
+            result.close()
+            return MessageHistory(*ranges, first=first)
+        else:
+            return None
+
+    def insert_message_hist(self, trans, channel, mhist):
```

Comment:
is this function actually worth having? it is only ever called to insert blank histories. `message_hist_values` will always return
```python
{
    'channel_id': channel.id,
    'first_message_id': None,
    'start_ranges': [],
    'end_ranges': [],
}
```

Reply:
It's also invoked when channels are created or destroyed.

Reply:
yeah, but isn't it called with a blank `mhist`? (my question is also about `insert_message_hist`)

Reply:
Oh huh, you're right. Yeah I guess that makes sense now, since you call `update_message_hist` if you want to modify it.
Yeah I'll change this.